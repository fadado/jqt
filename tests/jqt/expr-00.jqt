=======================
Tests about expressions
=======================

Escape: \\ TB (\t) LF(\n) CR(\r)

Interpolated string: {{.yes}}
Interpolated string: {{.not}}
Interpolated string: {{.nil}}
Interpolated string: {{.string}}
Interpolated string: {{.number}}

_______________________
Vanished line: {{empty}}
Vanished line: {{.not//empty}}
Vanished line: {{.nil//empty}}
_______________________

Not vanished line: {{.yes//empty}}
Not vanished line: {{.string//empty}}
Not vanished line: {{.number//empty}}
Not vanished line: {{.array//empty}}
Not vanished line: {{.object//empty}}

Nested expression: {{if .yes then "==={{.number}}===" else "==={{.string}}===" end}}
Nested expression: {{if .not then "==={{.number}}===" else "==={{.string}}===" end}}
Nested expression: {{"={{"-{{"*"}}-"}}="}}

Nested expression: {{
	if .yes
	then "===\(.number)==="
	else "===\(.string)==="
	end
}}
Nested expression: {{
	if .not
	then "===\(.number)==="
	else "===\(.string)==="
	end
}}
Nested expression: {{
	"=\(
		"-\(
			"*"
		)-"
	)="
}}

Local name: {{. as $jqt | $jqt.number}}
Local name: {{. as $jqt | $jqt.array | map(.*2) | sort | map(tostring)}}
Local name: {{. as $jqt | $jqt.array | map(.*2) | map(tostring) | sort}}

Repeated line: {{.array | keys[]}}
Repeated line: {{.object | keys[]}}
Repeated line: {{.array[]}}

Repeated line: {{. as $jqt | range(1;5) | .*$jqt.number}}

Repeated line: {{.array | sort[]}}

Repeated line: {{
	if .nil == null
	then .array | sort[]
	else empty end
}}

{% .number %}The value of pi is {{.}}.
{% .number %}Pi squared: {{.*2}}
{% .number*2 %}Pi squared: {{.}}
{% .array %}Array {{.}} has length {{.|length}}
{% .array %}Array {{.}} has length {{length}}

{% .array as $a %}Array $a has value {{$a[0]}} at position 0
{% .array as $a | $jqt %}Array .array and array $a are equal? ({{$a == $jqt.array)}}
{% .array | map(.*2 | tostring) | sort %}Values sorted as strings: {{.}}

{% .array | sort[] | tostring | "<b>{{.}}</b>" | "<li>{{.}}</li>" %}{{.}}

Slice: {{ (.array | sort | reverse)[:4]}}
Limit: {{[limit(4; .array | sort | reverse[])]}}

Modulo: {{ 2 % 7 }}
{% 2 % 7 %}Modulo: {{.}}

Object: {{ {a:1}+{b:2} }}

Fibonacci(10): {{ [limit(10; fibonacci)] }}
Fibonacci(10): {{ [limit(10; fibonacci)][-1] }}
Fibonacci(10): {{ [limit(10; fibonacci)] | reverse[0] }}

=======================
