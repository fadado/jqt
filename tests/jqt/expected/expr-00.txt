=======================
Tests about expressions
=======================

Escape: \\ TB (\t) LF(\n) CR(\r)

Interpolated string: true
Interpolated string: false
Interpolated string: null
Interpolated string: Neque porro quisquam
Interpolated string: 3.141592653589793

_______________________
_______________________

Not vanished line: true
Not vanished line: Neque porro quisquam
Not vanished line: 3.141592653589793
Not vanished line: [7,6,5,4,3,2,1]
Not vanished line: {"b":true,"s":"Neque","n":3.14}

Nested expression: ===3.141592653589793===
Nested expression: ===Neque porro quisquam===
Nested expression: =-*-=

Nested expression: ===3.141592653589793===
Nested expression: ===Neque porro quisquam===
Nested expression: =-*-=

Local name: 3.141592653589793
Local name: ["2","4","6","8","10","12","14"]
Local name: ["10","12","14","2","4","6","8"]

Repeated line: 0
Repeated line: 1
Repeated line: 2
Repeated line: 3
Repeated line: 4
Repeated line: 5
Repeated line: 6
Repeated line: b
Repeated line: n
Repeated line: s
Repeated line: 7
Repeated line: 6
Repeated line: 5
Repeated line: 4
Repeated line: 3
Repeated line: 2
Repeated line: 1

Repeated line: 3.141592653589793
Repeated line: 6.283185307179586
Repeated line: 9.42477796076938
Repeated line: 12.566370614359172

Repeated line: 1
Repeated line: 2
Repeated line: 3
Repeated line: 4
Repeated line: 5
Repeated line: 6
Repeated line: 7

Repeated line: 1
Repeated line: 2
Repeated line: 3
Repeated line: 4
Repeated line: 5
Repeated line: 6
Repeated line: 7

The value of pi is 3.141592653589793.
Pi squared: 6.283185307179586
Pi squared: 6.283185307179586
Array [7,6,5,4,3,2,1] has length 7
Array [7,6,5,4,3,2,1] has length 7

Array $a has value 7 at position 0
Array .array and array $a are equal? (true)
Values sorted as strings: ["10","12","14","2","4","6","8"]

<li><b>1</b></li>
<li><b>2</b></li>
<li><b>3</b></li>
<li><b>4</b></li>
<li><b>5</b></li>
<li><b>6</b></li>
<li><b>7</b></li>

Slice: [7,6,5,4]
Limit: [7,6,5,4]

Modulo: 2
Modulo: 2

Object: {"a":1,"b":2}

Fibonacci(10): [0,1,1,2,3,5,8,13,21,34]
Fibonacci(10): 34
Fibonacci(10): 34

=======================

